### PR TITLE
Fix tests in org.eclipse.core.expressions.tests #525

### PR DESCRIPTION
**Pull request by Michael5601**
This commit fixes the tests testLessThanOrEqualToExpression and testGreaterThanOrEqualToExpression of class CountExpression.java in org.eclipse.core.expressions.tests by adding additional functionality similar to the already existing. Contributes to #525.

### CODE REVIEW

**Comment by Michael5601** [comment]
Failing test documented in https://github.com/eclipse-platform/eclipse.platform/issues/628.

File: runtime/bundles/org.eclipse.core.expressions/src/org/eclipse/core/expressions/CountExpression.java [hunk 1]
```diff
@@ -62,68 +63,86 @@ public CountExpression(String size) {
...
 	}
 
 	@Override
 	public EvaluationResult evaluate(IEvaluationContext context) throws CoreException {
-		Object var= context.getDefaultVariable();
+		Object var = context.getDefaultVariable();
 		int size;
 		if (var instanceof Collection) {
-			size= ((Collection<?>)var).size();
+			size = ((Collection<?>) var).size();
```
Comment (HeikoKlare) [inline runtime/bundles/org.eclipse.core.expressions/src/org/eclipse/core/expressions/CountExpression.java:118]: Please replace the cast with an assignment in the condition, i.e.:
```java
if (var instanceof Collection collection) {
	size = collection.size();
```

**Review by HeikoKlare — COMMENTED** [review]
Thanks for adding the missing functionality and enabling the according tests!

The changes look sound to me. Tests work as expected. I would only suggest to apply a minor refactoring moving to recent Java features, since the code is touched anyway (see code comment).

File: runtime/bundles/org.eclipse.core.expressions/src/org/eclipse/core/expressions/CountExpression.java (hunk 1 above)
Comment (Michael5601) [reply runtime/bundles/org.eclipse.core.expressions/src/org/eclipse/core/expressions/CountExpression.java]: Thank you for for your reply!

I added the code refactoring in the commit. The tests will run again now and if there is no failing test, the PR can be merged.

**Review by HeikoKlare — APPROVED** [review]
With recent improvements, changes are fine for me.
I added the required version bumps for 4.30.

**Pull request merged**